recipes/proposal/all/conanfile.py

```diff
@@ -0,0 +1,98 @@
+from conans import ConanFile, CMake, tools
+from conans.errors import ConanInvalidConfiguration
+import os
+
+required_conan_version = ">=1.32.0"
+
+
+class PROPOSALConan(ConanFile):
+    name = "proposal"
+    homepage = "https://github.com/tudo-astroparticlephysics/PROPOSAL"
+    license = "GNU Lesser General Public License v3.0"
+    url = "https://github.com/conan-io/conan-center-index"
+    description = "monte Carlo based lepton and photon propagator"
+    topics = ("propagator", "lepton", "photon", "stochastic")
+
+    settings = "os", "compiler", "build_type", "arch"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+        "build_testing": [True, False],
+        "build_python": [True, False],
+        "build_documentation": [True, False],
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+        "build_testing": False,
+        "build_python": False,
+        "build_documentation": False,
+    }
+
+    generators = "cmake_find_package", "cmake_paths"
```

Comment:
I am not completely aware if I get the point with the cmake wrapper.

Do you mean these [cmake-conan](https://github.com/conan-io/cmake-conan) wrapper?

The use case we want to provide, that there are some user on system which aren't able to or won't install this lib with conan, whyever. Therefore, we have to use the `find_package` function to find the local installed packages. If I naively use the `cmake` generator the `find_package` functions won't work if I haven't missed something. 

I have already asked if it's okay to use the `cmake_find_package` generator https://github.com/conan-io/conan-center-index/pull/4695#discussion_r583802351. Why isn't it here possible?

Reply:
Don't use conan-cmake, that's a whole other can of worms.
I meant the [cmake generator](https://docs.conan.io/en/latest/reference/generators/cmake.html).

You can check other recipes in this repo on how to do this.

Reply:
Does cdc8b0e65f32f947af62c7b69c262db5a3647069 fix this issue? :sweat_smile:

Reply:
Not completely, you need to add a cmake wrapper such as https://github.com/conan-io/conan-center-index/blob/master/recipes/aws-c-common/all/CMakeLists.txt
This file has to be added to `export_sources` + the call to `cmake.configure` needs to refer to the wrapper `CMakeLists.txt` instead of the one in `source_subfolder`.